xbmc/addons/interfaces/Filesystem.cpp

```diff
@@ -551,32 +748,33 @@ char** Interface_Filesystem::get_property_values(
   XFILE::FileProperty internalType;
   switch (type)
   {
-  case ADDON_FILE_PROPERTY_RESPONSE_PROTOCOL:
-    internalType = XFILE::FILE_PROPERTY_RESPONSE_PROTOCOL;
-    break;
-  case ADDON_FILE_PROPERTY_RESPONSE_HEADER:
-    internalType = XFILE::FILE_PROPERTY_RESPONSE_HEADER;
-    break;
-  case ADDON_FILE_PROPERTY_CONTENT_TYPE:
-    internalType = XFILE::FILE_PROPERTY_CONTENT_TYPE;
-    break;
-  case ADDON_FILE_PROPERTY_CONTENT_CHARSET:
-    internalType = XFILE::FILE_PROPERTY_CONTENT_CHARSET;
-    break;
-  case ADDON_FILE_PROPERTY_MIME_TYPE:
-    internalType = XFILE::FILE_PROPERTY_MIME_TYPE;
-    break;
-  case ADDON_FILE_PROPERTY_EFFECTIVE_URL:
-    internalType = XFILE::FILE_PROPERTY_EFFECTIVE_URL;
-    break;
-  default:
-    CLog::Log(LOGERROR, "Interface_Filesystem::{} - invalid data (addon='{}', file='{}')",
-              __FUNCTION__, kodiBase, file);
-    return nullptr;
+    case ADDON_FILE_PROPERTY_RESPONSE_PROTOCOL:
+      internalType = XFILE::FILE_PROPERTY_RESPONSE_PROTOCOL;
+      break;
+    case ADDON_FILE_PROPERTY_RESPONSE_HEADER:
+      internalType = XFILE::FILE_PROPERTY_RESPONSE_HEADER;
+      break;
+    case ADDON_FILE_PROPERTY_CONTENT_TYPE:
+      internalType = XFILE::FILE_PROPERTY_CONTENT_TYPE;
+      break;
+    case ADDON_FILE_PROPERTY_CONTENT_CHARSET:
+      internalType = XFILE::FILE_PROPERTY_CONTENT_CHARSET;
+      break;
+    case ADDON_FILE_PROPERTY_MIME_TYPE:
+      internalType = XFILE::FILE_PROPERTY_MIME_TYPE;
+      break;
+    case ADDON_FILE_PROPERTY_EFFECTIVE_URL:
+      internalType = XFILE::FILE_PROPERTY_EFFECTIVE_URL;
+      break;
+    default:
+      CLog::Log(LOGERROR, "Interface_Filesystem::{} - invalid data (addon='{}', file='{}')",
+                __FUNCTION__, kodiBase, file);
+      return nullptr;
   };
-  std::vector<std::string> values = static_cast<CFile*>(file)->GetPropertyValues(internalType, name);
+  std::vector<std::string> values =
+      static_cast<CFile*>(file)->GetPropertyValues(internalType, name);
   *numValues = values.size();
-  char **ret = static_cast<char**>(malloc(sizeof(char*)*values.size()));
+  char** ret = static_cast<char**>(malloc(sizeof(char*) * values.size()));
```

Comment:
Another opportunity for `new`? I saw you made such a change above, just wondering

Reply:
Makes sense to use `new`, place there was come by clang cleanup and overseen it.

Must only learn how is on `**` memory :-).

Reply:
Is then `char** ret = new char*[size];` for there, but can't used as his counterpart use `free()`
Here: https://github.com/xbmc/xbmc/blob/master/xbmc/addons/interfaces/AddonBase.cpp#L455-L465